iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -370,29 +383,48 @@ cxx::expected<int32_t, IpcChannelError> UnixDomainSocket::createSocket(const Ipc
         }
         else
         {
+            auto closeCall = closeFd(sockfd);
+            if (closeCall.has_error())
+            {
+                return cxx::error<IpcChannelError>(closeCall.get_error());
+            }
             return createErrorFromErrnum(bindCall.getErrNum());
         }
     }
     else
     {
-        // we use a connected socket, this leads to a behavior closer to the message queue (e.g. error if client is
-        // created and server not present)
+        // we use a connected socket, this leads to a behavior closer to the message queue (e.g. error if client
+        // is created and server not present)
         auto connectCall = cxx::makeSmartC(connect,
                                            cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
                                            {ERROR_CODE},
-                                           {},
+                                           {ENOENT},
                                            sockfd,
                                            (struct sockaddr*)&m_sockAddr,
                                            static_cast<socklen_t>(sizeof(m_sockAddr)));
 
-        if (!connectCall.hasErrors())
+        if (connectCall.hasErrors())
         {
-            return cxx::success<int32_t>(sockfd);
+            auto closeCall = closeFd(sockfd);
+            if (closeCall.has_error())
+            {
+                return cxx::error<IpcChannelError>(closeCall.get_error());
+            }
+            return createErrorFromErrnum(connectCall.getErrNum());
         }
-        else
+        else if (connectCall.getErrNum() == ENOENT)
         {
+            auto closeCall = closeFd(sockfd);
```

Comment:
We have this code snippet at least 3 times in the code here:
```cpp
            auto closeCall = closeFd(sockfd);
            if (closeCall.has_error())
            {
                return cxx::error<IpcChannelError>(closeCall.get_error());
            }
```

I would suggest a lambda like
```cpp
auto callClose = [this](){
            auto closeCall = closeFd(sockfd);
            if (closeCall.has_error())
            {
                return cxx::error<IpcChannelError>(closeCall.get_error());
            }
};

///....

return callClose();
```